src/System.Device.Gpio/build/net5.0/System.Device.Gpio.targets

```diff
@@ -0,0 +1,7 @@
+<Project>
+  <Target Name="EnsureProjectIsTargetingWindowsPlatformIfRequired"
+          BeforeTargets="Build"
+          Condition="'$(TargetFramework)' == 'net5.0' And $(RuntimeIdentifier.StartsWith('win'))">
+    <Error Text="In order to use System.Device.Gpio package when running on Windows and targeting net5.0, it is required to add 'net5.0-windows10.0.17763.0' to your project's target frameworks." />
```

Comment:
It doesn't have to be exactly 17763, it could be that version or higher.  Also, I'm not sure if the RuntimeIdentifier will necessarily be set in all scenarios where you'd want this check.

Reply:
I'll change the message to say that 17763 is just the lower bound. What would you recommend instead of using RuntimeIdentifier? I know that we might still have some cases where we won't be able to catch problems, like for example shared framework applications, but that's why for those we are relying on the PlatformNotSupported exception messages to help figure out how to fix the targeting. Of course, if there are better ways to catch it at compile time instead of runtime that would be much prefered.

Reply:
Not sure this should cause an error at all. Even if I'm running on windows, I might not need to use the windows specific implementation, especially if I'm not using the WindowsGpioDriver (but maybe only an Arduino driver).

Reply:
While most (if not all) of this library would render useless if we can call WinRT in Windows, I do agree with you that perhaps Error is too aggressive even though it is suppressable. @ericstj do you think switching to Warning would be appropriate here?

Reply:
Chatted offline, I'll keep as error for now since we want to make sure that users won't miss it, but keep it suppressible so that users can disable if they think they are a false positive.